dagster_rudderstack/resources/rudderstack.py

```
@@ -0,0 +1,210 @@
+    )
+    poll_timeout: float = Field(
+        default=None,
+        description="Time (in seconds) after which the polling for a triggered job is declared timed out.",
+    )
+
+    @property
+    @abstractmethod
+    def api_base_url(self) -> str:
+        raise NotImplementedError()
+
+    @property
+    @abstractmethod
+    def request_headers(self) -> Dict[str, any]:
+        raise NotImplementedError()
+
+    @property
+    @cached_method
+    def _log(self) -> logging.Logger:
+        return get_dagster_logger()
+
+    def make_request(
+        self,
+        endpoint: str,
+        method: str = "POST",
+        params: Optional[Dict[str, Any]] = None,
+        data: Optional[Mapping[str, object]] = None,
+    ):
+        """Prepares and makes request to RudderStack API endpoint.
+
+        Args:
+            method (str): The http method to be used for this request (e.g. "GET", "POST").
+            endpoint (str): The RudderStack API endpoint to send request to.
+            params (Optional(dict)): Query parameters to pass to the API endpoint
+
+        Returns:
+            Dict[str, Any]: Parsed json data from the response for this request.
+        """
+        url = urljoin(self.api_base_url, endpoint)
+        headers = self.request_headers
+        num_retries = 0
+
+        while True:
+            try:
+                request_args: Dict[str, Any] = dict(
+                    method=method,
+                    url=url,
+                    headers=headers,
+                    timeout=self.request_timeout,
+                )
+                if data:
+                    request_args["json"] = data
+                response = requests.request(**request_args)
+                response.raise_for_status()
+                return response.json()
+            except requests.RequestException as e:
+                self._log.error(f"Request to url: {url} failed: {e}")
+                if num_retries == self.request_max_retries:
```

is there option for this to be infinite retry if customer doesn't want any ? could be default behavior

---

`is there option for this to be infinite retry if customer doesn't want any` - No
Not sure if this is right to keep retries on error infinite. Better to let it fail after few retries and we can get issue from customer.